MREUnityRuntime/MREUnityRuntimeLib/Util/Unity/AssetFetcher.cs

```diff
@@ -67,13 +81,16 @@ IEnumerator LoadCoroutine()
 					{
 						if (typeof(T) == typeof(AudioClip))
 						{
-							result.Asset = ((DownloadHandlerAudioClip) handler).audioClip as T;
+							result.Asset = ((DownloadHandlerAudioClip)handler).audioClip as T;
 						}
 						else if (typeof(T) == typeof(Texture))
 						{
-							result.Asset = ((DownloadHandlerTexture) handler).texture as T;
+							result.Asset = ((DownloadHandlerTexture)handler).texture as T;
 						}
 					}
+
+					// Decrement activeLoads
+					Interlocked.Decrement(ref activeLoads);
```

Comment:
Would be nice to be able to decrement the counter before the finalization handler was called.